src/scripts/main.js

```diff
@@ -1,3 +1,301 @@
+
+  label.textContent = formatLabelText(inputName);
+  label.append(input);
+
+  return label;
+};
+
+const createSelect = (inputName, options) => {
+  const select = Object.assign(document.createElement('select'), {
+    name: inputName,
+  });
+
+  select.setAttribute('data-qa', inputName);
+
+  Array.from(options)
+    .sort()
+    .forEach((option) => select.append(new Option(option, option)));
+
+  const label = document.createElement('label');
+
+  label.textContent = formatLabelText(inputName);
+  label.append(select);
+
+  return label;
+};
+
+const createButton = (btnText, type) => {
+  const button = document.createElement('button');
+
+  button.setAttribute('type', type);
+  button.textContent = btnText;
+
+  return button;
+};
+
+const form = document.createElement('form');
+const offices = [
+  'Tokyo',
+  'Singapore',
+  'London',
+  'New York',
+  'Edinburgh',
+  'San Francisco',
+];
+
+const formElements = [
+  createInput('name', 'text'),
+  createInput('position', 'text'),
+  createSelect('office', offices),
+  createInput('age', 'number'),
+  createInput('salary', 'number'),
+  createButton('Save to table', 'submit'),
+];
+
+form.classList.add('new-employee-form');
+formElements.forEach((element) => form.append(element));
+document.addEventListener('DOMContentLoaded', () => document.body.append(form));
+
+// FORM VALIDATION
+const validateFormInputData = (value, key) => {
+  const validators = {
+    position: () =>
+      !value
+        ? { validated: false, message: `Position is a required field!` }
+        : { validated: true },
+    name: () =>
+      !value || value.length < 4
+        ? { validated: false, message: 'Name should be more than 4 chars long' }
+        : { validated: true },
+    age: () =>
+      value < 18 || value > 90
+        ? { validated: false, message: 'Age must be between 18 and 90' }
+        : { validated: true },
+  };
+
+  return validators[key]?.() || { validated: true };
+};
+
+const validateForm = (data) => {
+  return Array.from(data)
+    .map(([key, value]) => validateFormInputData(value, key))
+    .filter((result) => !result.validated)
+    .map((result) => result.message);
+};
+
+// NOTIFICATIONS ON SUCESS / ERROR
+const displaySuccessNotice = (message) => {
+  displayNotification('success', 'Success!', [message]);
+};
+
+const displayErrorNotice = (errorArray) => {
+  displayNotification('error', 'Error!', errorArray);
+};
+
+const displayNotification = (type, titleText, errorMessageArr) => {
+  const notification = document.createElement('div');
+
+  notification.classList.add('notification', type);
+  notification.setAttribute('data-qa', 'notification');
+
+  notification.innerHTML = `
+    <h3 class="title">${titleText}</h3>
+    <p>
+      <ul>
+        ${errorMessageArr.map((message) => `<li>${message}</li>`).join('')}
+      </ul>
+    </p>
+  `;
+
+  document.body.append(notification);
+  setTimeout(() => notification.remove(), 2000);
+};
+
```

Comment:
The `numToSalary` function is a utility function that could be used in other parts of your codebase. Consider moving it to a separate utilities file.